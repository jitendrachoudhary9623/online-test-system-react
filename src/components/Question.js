import React,{Component} from 'react';

class Question extends Component{

    constructor(props){
        super(props);
        }

    handleAnswerLayoutChange(answerNo){
        this.props.handleAnswerClick(this.props.item.id,answerNo);
    }
    render(){
        return(
            <ul key={this.props.id} className="card questionBody" >
            <li className="card-content">
            {this.props.item.id} . {this.props.item.Question}
            </li>
            <li>
                {this.props.item.Answers.map((ans, i) => (
                    <ul className="card-action" key={'' + this.props.item.id + i + 1}>
                        <li onClick={this.handleAnswerLayoutChange.bind(this, i + 1)}>
                            <div className={ans.marked === true ? "btn blue darken-1" : "btn blue-grey darken-5"}>
                                {i + 1} . {ans.option}
                            </div>
                        </li>
                    </ul>
                ))}
            </li>
        </ul>
        );
    }
}

export default Question;
