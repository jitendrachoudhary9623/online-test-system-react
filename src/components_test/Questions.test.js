import React, { Component } from 'react';
import Question from "../components/Question"
import { shallow } from 'enzyme'

describe('UI Test for Question', () => {

    it('Question to be defined', () => {
        expect(Question).toBeDefined();
    });

    it('Question should render correctly', () => {
        const mockHandler = jest.fn(event => {
            console.log("Mocked function");
        });
        let questionMock = {
            "id": 1,
            "type": "objective",
            "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
            "Answers": [{ option: "6.54 sec", marked: false }, { option: "44.32 sec", marked: false }, { option: "55 sec", marked: false }, { option: "30.2 sec", marked: false }],
            "userAnswer": "",
            "actualAnswer": "1"
        };
        const snapshot = shallow(
            <Question key={'b1'} item={questionMock} id={'a1'} handleAnswerClick={mockHandler} />
        );
        expect(snapshot).toMatchSnapshot();
    });

    it('Question should render 4 answer box for 4 number of choices  ', () => {
        const mockHandler = jest.fn(event => {
            console.log("Mocked function");
        });
        let questionMock = {
            "id": 1,
            "type": "objective",
            "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
            "Answers": [{ option: "6.54 sec", marked: false }, { option: "44.32 sec", marked: false }, { option: "55 sec", marked: false }, { option: "30.2 sec", marked: false }],
            "userAnswer": "",
            "actualAnswer": "1"
        };
        const snapshot = shallow(
            <Question key={'b1'} item={questionMock} id={'a1'} handleAnswerClick={mockHandler} />
        );
        expect(snapshot.find('.card-action').length).toBe(questionMock.Answers.length);
    });

    it('Question should render 2 answer box for 2 number of choices  ', () => {
        const mockHandler = jest.fn(event => {
            console.log("Mocked function");
        });
        let questionMock = {
            "id": 1,
            "type": "objective",
            "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
            "Answers": [{ option: "55 sec", marked: false }, { option: "30.2 sec", marked: false }],
            "userAnswer": "",
            "actualAnswer": "1"
        };
        const snapshot = shallow(
            <Question key={'b1'} item={questionMock} id={'a1'} handleAnswerClick={mockHandler} />
        );
        expect(snapshot.find('.card-action').length).toBe(questionMock.Answers.length);
    });


});