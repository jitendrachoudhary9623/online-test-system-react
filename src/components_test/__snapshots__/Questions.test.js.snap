exports[`UI Test for Question Question should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    handleAnswerClick={[Function]}
    id="a1"
    item={
        Object {
            "Answers": Array [
              Object {
                "marked": false,
                "option": "6.54 sec",
              },
              Object {
                "marked": false,
                "option": "44.32 sec",
              },
              Object {
                "marked": false,
                "option": "55 sec",
              },
              Object {
                "marked": false,
                "option": "30.2 sec",
              },
            ],
            "Question": "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
            "actualAnswer": "1",
            "id": 1,
            "type": "objective",
            "userAnswer": "",
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "a1",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="card-content">
          1
           . 
          A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?
</li>,
        <li>
          <ul
                    className="card-action">
                    <li
                              onClick={[Function]}>
                              <div
                                        className="btn blue-grey darken-5">
                                        1
                                         . 
                                        6.54 sec
                              </div>
                    </li>
          </ul>
          <ul
                    className="card-action">
                    <li
                              onClick={[Function]}>
                              <div
                                        className="btn blue-grey darken-5">
                                        2
                                         . 
                                        44.32 sec
                              </div>
                    </li>
          </ul>
          <ul
                    className="card-action">
                    <li
                              onClick={[Function]}>
                              <div
                                        className="btn blue-grey darken-5">
                                        3
                                         . 
                                        55 sec
                              </div>
                    </li>
          </ul>
          <ul
                    className="card-action">
                    <li
                              onClick={[Function]}>
                              <div
                                        className="btn blue-grey darken-5">
                                        4
                                         . 
                                        30.2 sec
                              </div>
                    </li>
          </ul>
</li>,
      ],
      "className": "card questionBody",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            " . ",
            "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
          ],
          "className": "card-content",
        },
        "ref": null,
        "rendered": Array [
          1,
          " . ",
          "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="card-action">
              <li
                            onClick={[Function]}>
                            <div
                                          className="btn blue-grey darken-5">
                                          1
                                           . 
                                          6.54 sec
                            </div>
              </li>
</ul>,
            <ul
              className="card-action">
              <li
                            onClick={[Function]}>
                            <div
                                          className="btn blue-grey darken-5">
                                          2
                                           . 
                                          44.32 sec
                            </div>
              </li>
</ul>,
            <ul
              className="card-action">
              <li
                            onClick={[Function]}>
                            <div
                                          className="btn blue-grey darken-5">
                                          3
                                           . 
                                          55 sec
                            </div>
              </li>
</ul>,
            <ul
              className="card-action">
              <li
                            onClick={[Function]}>
                            <div
                                          className="btn blue-grey darken-5">
                                          4
                                           . 
                                          30.2 sec
                            </div>
              </li>
</ul>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "101",
            "nodeType": "host",
            "props": Object {
              "children": <li
                onClick={[Function]}>
                <div
                                className="btn blue-grey darken-5">
                                1
                                 . 
                                6.54 sec
                </div>
</li>,
              "className": "card-action",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="btn blue-grey darken-5">
                  1
                   . 
                  6.54 sec
</div>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    1,
                    " . ",
                    "6.54 sec",
                  ],
                  "className": "btn blue-grey darken-5",
                },
                "ref": null,
                "rendered": Array [
                  1,
                  " . ",
                  "6.54 sec",
                ],
                "type": "div",
              },
              "type": "li",
            },
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": "111",
            "nodeType": "host",
            "props": Object {
              "children": <li
                onClick={[Function]}>
                <div
                                className="btn blue-grey darken-5">
                                2
                                 . 
                                44.32 sec
                </div>
</li>,
              "className": "card-action",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="btn blue-grey darken-5">
                  2
                   . 
                  44.32 sec
</div>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    2,
                    " . ",
                    "44.32 sec",
                  ],
                  "className": "btn blue-grey darken-5",
                },
                "ref": null,
                "rendered": Array [
                  2,
                  " . ",
                  "44.32 sec",
                ],
                "type": "div",
              },
              "type": "li",
            },
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": "121",
            "nodeType": "host",
            "props": Object {
              "children": <li
                onClick={[Function]}>
                <div
                                className="btn blue-grey darken-5">
                                3
                                 . 
                                55 sec
                </div>
</li>,
              "className": "card-action",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="btn blue-grey darken-5">
                  3
                   . 
                  55 sec
</div>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    3,
                    " . ",
                    "55 sec",
                  ],
                  "className": "btn blue-grey darken-5",
                },
                "ref": null,
                "rendered": Array [
                  3,
                  " . ",
                  "55 sec",
                ],
                "type": "div",
              },
              "type": "li",
            },
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": "131",
            "nodeType": "host",
            "props": Object {
              "children": <li
                onClick={[Function]}>
                <div
                                className="btn blue-grey darken-5">
                                4
                                 . 
                                30.2 sec
                </div>
</li>,
              "className": "card-action",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="btn blue-grey darken-5">
                  4
                   . 
                  30.2 sec
</div>,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    4,
                    " . ",
                    "30.2 sec",
                  ],
                  "className": "btn blue-grey darken-5",
                },
                "ref": null,
                "rendered": Array [
                  4,
                  " . ",
                  "30.2 sec",
                ],
                "type": "div",
              },
              "type": "li",
            },
            "type": "ul",
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "a1",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="card-content">
            1
             . 
            A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?
</li>,
          <li>
            <ul
                        className="card-action">
                        <li
                                    onClick={[Function]}>
                                    <div
                                                className="btn blue-grey darken-5">
                                                1
                                                 . 
                                                6.54 sec
                                    </div>
                        </li>
            </ul>
            <ul
                        className="card-action">
                        <li
                                    onClick={[Function]}>
                                    <div
                                                className="btn blue-grey darken-5">
                                                2
                                                 . 
                                                44.32 sec
                                    </div>
                        </li>
            </ul>
            <ul
                        className="card-action">
                        <li
                                    onClick={[Function]}>
                                    <div
                                                className="btn blue-grey darken-5">
                                                3
                                                 . 
                                                55 sec
                                    </div>
                        </li>
            </ul>
            <ul
                        className="card-action">
                        <li
                                    onClick={[Function]}>
                                    <div
                                                className="btn blue-grey darken-5">
                                                4
                                                 . 
                                                30.2 sec
                                    </div>
                        </li>
            </ul>
</li>,
        ],
        "className": "card questionBody",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              " . ",
              "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
            ],
            "className": "card-content",
          },
          "ref": null,
          "rendered": Array [
            1,
            " . ",
            "A train 120 meters long is running with a speed of 60 km/hr. In what time will it pass a boy who is running at 6 km/hr in the direction opposite to that in which the train is going?",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                className="card-action">
                <li
                                onClick={[Function]}>
                                <div
                                                className="btn blue-grey darken-5">
                                                1
                                                 . 
                                                6.54 sec
                                </div>
                </li>
</ul>,
              <ul
                className="card-action">
                <li
                                onClick={[Function]}>
                                <div
                                                className="btn blue-grey darken-5">
                                                2
                                                 . 
                                                44.32 sec
                                </div>
                </li>
</ul>,
              <ul
                className="card-action">
                <li
                                onClick={[Function]}>
                                <div
                                                className="btn blue-grey darken-5">
                                                3
                                                 . 
                                                55 sec
                                </div>
                </li>
</ul>,
              <ul
                className="card-action">
                <li
                                onClick={[Function]}>
                                <div
                                                className="btn blue-grey darken-5">
                                                4
                                                 . 
                                                30.2 sec
                                </div>
                </li>
</ul>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "101",
              "nodeType": "host",
              "props": Object {
                "children": <li
                  onClick={[Function]}>
                  <div
                                    className="btn blue-grey darken-5">
                                    1
                                     . 
                                    6.54 sec
                  </div>
</li>,
                "className": "card-action",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="btn blue-grey darken-5">
                    1
                     . 
                    6.54 sec
</div>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      1,
                      " . ",
                      "6.54 sec",
                    ],
                    "className": "btn blue-grey darken-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    1,
                    " . ",
                    "6.54 sec",
                  ],
                  "type": "div",
                },
                "type": "li",
              },
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": "111",
              "nodeType": "host",
              "props": Object {
                "children": <li
                  onClick={[Function]}>
                  <div
                                    className="btn blue-grey darken-5">
                                    2
                                     . 
                                    44.32 sec
                  </div>
</li>,
                "className": "card-action",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="btn blue-grey darken-5">
                    2
                     . 
                    44.32 sec
</div>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      2,
                      " . ",
                      "44.32 sec",
                    ],
                    "className": "btn blue-grey darken-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    2,
                    " . ",
                    "44.32 sec",
                  ],
                  "type": "div",
                },
                "type": "li",
              },
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": "121",
              "nodeType": "host",
              "props": Object {
                "children": <li
                  onClick={[Function]}>
                  <div
                                    className="btn blue-grey darken-5">
                                    3
                                     . 
                                    55 sec
                  </div>
</li>,
                "className": "card-action",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="btn blue-grey darken-5">
                    3
                     . 
                    55 sec
</div>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      3,
                      " . ",
                      "55 sec",
                    ],
                    "className": "btn blue-grey darken-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    3,
                    " . ",
                    "55 sec",
                  ],
                  "type": "div",
                },
                "type": "li",
              },
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": "131",
              "nodeType": "host",
              "props": Object {
                "children": <li
                  onClick={[Function]}>
                  <div
                                    className="btn blue-grey darken-5">
                                    4
                                     . 
                                    30.2 sec
                  </div>
</li>,
                "className": "card-action",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="btn blue-grey darken-5">
                    4
                     . 
                    30.2 sec
</div>,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      4,
                      " . ",
                      "30.2 sec",
                    ],
                    "className": "btn blue-grey darken-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    4,
                    " . ",
                    "30.2 sec",
                  ],
                  "type": "div",
                },
                "type": "li",
              },
              "type": "ul",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
